adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapterTest.java

```diff
@@ -184,25 +197,52 @@ public void uploadTelemetryMessageWithUnsettledDeliverySemantics() {
         final MessageSender telemetrySender = givenATelemetrySenderForAnyTenant();
 
         // which is enabled for a tenant
-        givenAConfiguredTenant("enabled-tenant", true);
+        givenAConfiguredTenant(TEST_TENANT_ID, true);
 
         // IF a device send telemetry data (with un-settled delivery)
         final ProtonDelivery delivery = mock(ProtonDelivery.class);
         when(delivery.remotelySettled()).thenReturn(false);
-        final ResourceIdentifier resource = ResourceIdentifier.from("telemetry", "enabled-tenant", "the-device");
+        final ResourceIdentifier resource = ResourceIdentifier.from("telemetry", TEST_TENANT_ID, TEST_DEVICE);
         adapter.uploadMessage(new AmqpContext(delivery, getFakeMessage(), resource));
 
         // THEN the sender sends the message and waits for the outcome from the downstream peer
         verify(telemetrySender).sendAndWaitForOutcome(any(Message.class));
     }
 
+    /**
+     * Verifies that a request to upload a telemetry message from a device that belongs to a tenant for which the AMQP
+     * adapter is disabled fails and that the device is notified when the message cannot be processed (un-settled
+     * delivery).
+     * 
+     */
+    @Test
+    public void testUploadTelemetryMessageFailsForDisabledTenant() {
+        // GIVEN an adapter configured to use a user-define server.
+        final VertxBasedAmqpProtocolAdapter adapter = givenAnAmqpAdapter();
+        final MessageSender telemetrySender = givenATelemetrySenderForAnyTenant();
+
+        // AND given a tenant which is ENABLED but DISABLED for the AMQP Adapter
+        givenAConfiguredTenant(TEST_TENANT_ID, Boolean.FALSE);
+
+        // WHEN a device uploads telemetry data to the adapter (and wants to be notified of failure)
+        final ProtonDelivery delivery = mock(ProtonDelivery.class);
+        when(delivery.remotelySettled()).thenReturn(false);
+        final ResourceIdentifier resource = ResourceIdentifier.from("telemetry", TEST_TENANT_ID, TEST_DEVICE);
+        final AmqpContext context = spy(new AmqpContext(delivery, getFakeMessage(), resource));
+        adapter.uploadMessage(context);
+
+        // THEN the adapter does not send the message (regardless of the delivery mode).
+        verify(telemetrySender, never()).send(any(Message.class));
+        verify(telemetrySender, never()).sendAndWaitForOutcome(any(Message.class));
+
+        // AND notifies the device that the message failed to be processed (message was sent unsettled).
+        verify(context).handleFailure(any(ServiceInvocationException.class));
```

Comment:
IMHO you should verify that the delivery is rejected/released ...

Reply:
sure, thanks! will update PR soon